qiita_ware/commands.py

```diff
@@ -44,53 +42,27 @@ def submit_EBI(preprocessed_data_id, action, send, fastq_dir_fp=None):
     If fastq_dir_fp is passed, it must not contain any empty files, or
     gzipped empty files
     """
-
-    preprocessed_data = PreprocessedData(preprocessed_data_id)
-    preprocessed_data_id_str = str(preprocessed_data_id)
-    study = Study(preprocessed_data.study)
-    sample_template = SampleTemplate(study.sample_template)
-    prep_template = PrepTemplate(preprocessed_data.prep_template)
-
-    investigation_type = None
-    new_investigation_type = None
-
-    if send:
-        # If we intend actually to send the files, then change the status in
-        # the database
-        preprocessed_data.update_insdc_status('submitting')
-
-    if fastq_dir_fp is not None:
-        # If the user specifies a FASTQ directory, use it
-
-        # Set demux_samples to None so that MetadataTemplate.to_file will put
-        # all samples in the template files
-        demux_samples = None
-    else:
-        # If the user does not specify a FASTQ directory, create one and
-        # re-serialize the per-sample FASTQs from the demux file
-        fastq_dir_fp = mkdtemp(prefix=qiita_config.working_dir)
-        demux = [path for _, path, ftype in preprocessed_data.get_filepaths()
-                 if ftype == 'preprocessed_demux'][0]
-
-        # Keep track of which files were actually in the demux file so that we
-        # can write those rows to the prep and samples templates
-        demux_samples = set()
-
-        with open_file(demux) as demux_fh:
-            for samp, iterator in to_per_sample_ascii(demux_fh,
-                                                      list(sample_template)):
-                demux_samples.add(samp)
-                sample_fp = join(fastq_dir_fp, "%s.fastq.gz" % samp)
-                wrote_sequences = False
-                with gzopen(sample_fp, 'w') as fh:
-                    for record in iterator:
-                        fh.write(record)
-                        wrote_sequences = True
-
-                if not wrote_sequences:
-                    remove(sample_fp)
-
+    # step 1
+    ebi_submission = EBISubmission(preprocessed_data_id, action)
+
+    # step 2
+    ebi_submission.preprocessed_data.update_insdc_status('demuxing samples')
+    try:
+        demux_samples = ebi_submission.generate_demultiplexed_fastq()
+    except Exception as e:
+        if isdir(ebi_submission.ebi_dir):
+            rmtree(ebi_submission.ebi_dir)
+        ebi_submission.preprocessed_data.update_insdc_status(
+            'failed: %s' % str(e))
+        LogEntry.create('Runtime', str(e),
```

Comment:
Instead of just saving the exception message, I would save the whole traceback (for debugging purposes), seems to be something like this:

```
>>> from traceback import format_exc
>>> try:
...    1/0
... except:
...    x = format_exc()
... 
>>> print x
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
ZeroDivisionError: integer division or modulo by zero
```